rpc/tests/p2p.rs

```diff
@@ -0,0 +1,273 @@
+use crate::utils::client::{new_test_client, AuthLevel};
+use celestia_rpc::prelude::*;
+use celestia_types::p2p;
+use libp2p::{identity, PeerId};
+use tokio::time::{sleep, Duration};
+
+pub mod utils;
+
+async fn rpc_call_delay() {
+    // delay for RPC calls like connect/close to let node finish the operation before we query it
+    // again. Below 150 ms I start getting intermittent failures.
+    sleep(Duration::from_millis(150)).await;
+}
+
+#[tokio::test]
+async fn info_test() {
+    let client = new_test_client(AuthLevel::Admin).await.unwrap();
+    client.p2p_info().await.expect("Failed to get node info");
+}
+
+#[tokio::test]
+async fn add_remove_peer_test() {
+    // add and then remove a peer, testing outputs from `p2p.Peers` and `p2p.Connectedness`
+    let (addr_info, task) = utils::tiny_node::start_tiny_node()
+        .await
+        .expect("failed to spin up second node");
+    let client = new_test_client(AuthLevel::Admin).await.unwrap();
+
+    let initial_peers = client
+        .p2p_peers()
+        .await
+        .expect("failed to get initial peer list");
+    assert_eq!(initial_peers.contains(&addr_info.id), false);
```

Comment:
for the boolean asserts you should stick with `assert!(...)` and `assert!(!...)`. It's a bug that clippy passed on CI, please fix it with `cargo clippy --all --all-targets` and fix it's suggestions

Reply:
I had a sneaking suspicion that CI was going too smooth, thanks for noting the clippy bug.
Fixed `bool_assert_comparison` and other clippy errors.

Reply:
I meant to fix the CI too :sweat_smile:

Reply:
right, didn't realise that would also land in this PR :facepalm: 
added `--all-targets` flag to clippy CI config

Reply:
> would also land in this PR

I personally don't mind it as it's small fix and not in code directly, but if you prefer I'd happily accept another PR or ff merge after some later rebase